spring-kafka/src/main/java/org/springframework/kafka/support/serializer/DelegatingByTopicSerializer.java

```diff
@@ -0,0 +1,245 @@
+
+	/**
+	 * Set to false to make topic pattern matching case-insensitive.
+	 */
+	public static final String CASE_SENSITIVE = "spring.kafka.value.serialization.bytopic.case.insensitive";
+
+	private final Map<Pattern, Serializer<?>> delegates = new ConcurrentHashMap<>();
+
+	private boolean forKeys;
+
+	private boolean cased = true;
+
+	/**
+	 * Construct an instance that will be configured in {@link #configure(Map, boolean)}
+	 * with producer properties {@link #VALUE_SERIALIZATION_TOPIC_CONFIG} and
+	 * {@link #KEY_SERIALIZATION_TOPIC_CONFIG}.
+	 */
+	public DelegatingByTopicSerializer() {
+	}
+
+	/**
+	 * Construct an instance with the supplied mapping of topic patterns to delegate
+	 * serializers.
+	 * @param delegates the map of delegates.
+	 */
+	public DelegatingByTopicSerializer(Map<Pattern, Serializer<?>> delegates) {
+		this.delegates.putAll(delegates);
+	}
+
+	/**
+	 * Set to false to make topic name matching case insensitive.
+	 * @param caseSensitive false for case insensitive.
+	 */
+	public void setCaseSensitive(boolean caseSensitive) {
+		this.cased = caseSensitive;
+	}
+
+	@Override
+	public void configure(Map<String, ?> configs, boolean isKey) {
+		this.forKeys = isKey;
+		Object insensitive = configs.get(CASE_SENSITIVE);
+		if (insensitive instanceof String) {
+			this.cased = Boolean.parseBoolean((String) insensitive);
+		}
+		String configKey = configKey();
+		Object value = configs.get(configKey);
+		if (value == null) {
+			return;
+		}
+		else if (value instanceof Map) {
+			processMap(configs, isKey, configKey, value);
+		}
+		else if (value instanceof String) {
+			this.delegates.putAll(createDelegates((String) value, configs, isKey));
+		}
+		else {
+			throw new IllegalStateException(
+					configKey + " must be a map or String, not " + value.getClass());
+		}
+	}
+
+	@SuppressWarnings("unchecked")
+	private void processMap(Map<String, ?> configs, boolean isKey, String configKey, Object value) {
+		((Map<Object, Object>) value).forEach((key, serializer) -> {
+			Pattern pattern = obtainPattern(key);
+			if (serializer instanceof Serializer) {
+				this.delegates.put(pattern, (Serializer<?>) serializer);
+				((Serializer<?>) serializer).configure(configs, isKey);
+			}
+			else if (serializer instanceof Class) {
+				instantiateAndConfigure(configs, isKey, this.delegates, pattern, (Class<?>) serializer);
+			}
+			else if (serializer instanceof String) {
+				createInstanceAndConfigure(configs, isKey, this.delegates, pattern, (String) serializer);
+			}
+			else {
+				throw new IllegalStateException(configKey
+						+ " map entries must be Serializers or class names, not " + value.getClass());
+			}
+		});
+	}
+
+	private String configKey() {
+		return this.forKeys ? KEY_SERIALIZATION_TOPIC_CONFIG : VALUE_SERIALIZATION_TOPIC_CONFIG;
+	}
+
+	protected Map<Pattern, Serializer<?>> createDelegates(String mappings, Map<String, ?> configs,
```

Comment:
Another crazy thought: how about to have a super class (or implementation of both `(De)Serializer`) with all of these common API?

Reply:
Good suggestions, all.